superbench/benchmarks/micro_benchmarks/ib_validation_performance.py

```diff
@@ -0,0 +1,320 @@
+            type=str,
+            nargs='+',
+            default=['ib_write_bw'],
+            help='The ib command used to run, e.g., {}.'.format(' '.join(self.__support_ib_commands)),
+        )
+        self._parser.add_argument(
+            '--gid_index',
+            type=int,
+            default=0,
+            required=False,
+            help='Test uses GID with GID index taken from command.',
+        )
+        self._parser.add_argument(
+            '--pattern',
+            type=str,
+            default='one-to-one',
+            required=False,
+            help='Test IB traffic pattern type, e.g., {}.'.format(''.join(self.__patterns)),
+        )
+        self._parser.add_argument(
+            '--config',
+            type=str,
+            default=None,
+            required=False,
+            help='The path of config file on the target machines',
+        )
+        self._parser.add_argument(
+            '--bidirectional', action='store_true', default=False, help='Measure bidirectional bandwidth.'
+        )
+        self._parser.add_argument(
+            '--gpu_index', type=int, default=None, required=False, help='Test Use GPUDirect with the gpu index.'
+        )
+        self._parser.add_argument(
+            '--hostfile',
+            type=str,
+            default='/root/hostfile',
+            required=False,
+            help='The path of hostfile on the target machines',
+        )
+
+    def __one_to_many(self, n):
+        config = []
+        for client in range(n):
+            row = []
+            for server in range(n):
+                if server != client:
+                    pair = '{},{}'.format(server, client)
+                    row.append(pair)
+            row = ';'.join(row)
+            config.append(row)
+        return config
+
+    def __many_to_one(self, n):
+        config = []
+        for server in range(n):
+            row = []
+            for client in range(n):
+                if server != client:
+                    pair = '{},{}'.format(server, client)
+                    row.append(pair)
+            row = ';'.join(row)
+            config.append(row)
+        return config
+
+    def __roundRobin(self, candidates):
+        res = []
+        count = len(candidates)
+        robin = candidates
+        first = [candidates[0]]
+        for _ in range(count - 1):
+            pairs = [[robin[i], robin[count - i - 1]] for i in range(0, (count + 1) // 2)]
+            res.append(pairs)
+            robin = robin[1:]
+            robin = first + robin[1:] + robin[:1]
+        return res
+
+    def __fully_one_to_one(self, n):
+        config = []
+        if n % 2 == 1:
+            return config
+        candidates = list(range(n))
+        res = self.__roundRobin(candidates)
+        for line in res:
+            row = []
+            for pair in line:
+                row.append('{},{}'.format(pair[0], pair[1]))
+            row = ';'.join(row)
+            config.append(row)
```

Comment:
I think we can append directly in the `__roundRobin()` and remove `__roundRobin()` function directly. What's the benefit for this design? Maybe I don't get the key idea on this.